packs/duo/actions/auth.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2actions.runners.pythonrunner import Action
+import duo_client
+
+
+class Auth(Action):
+    def run(self, username, factor, ipaddr, async, device,
+            push_type, passcode, push_info):
+        """
+        Auth against the Duo Platorm.
+
+        Returns: An dict with info returned by Duo.
+
+        Raises:
+          ValueError: On Auth Failure.
+        """
+
+        try:
+            ikey = self.config['ikey']
+            skey = self.config['skey']
+            host = self.config['host']
+        except KeyError:
+            raise ValueError("Duo config not found in config.")
+
+        auth = duo_client.Auth(ikey=ikey,
+                               skey=skey,
+                               host=host)
+
+        try:
+            data = auth.auth(factor=factor,
+                             username=username,
+                             type=push_type,
+                             device=device)
+        except:
+            raise ValueError("Duo Auth request failed!")
+        else:
+            if data['status'] == "allow":
+                return data
+            elif data['status'] == "deny":
+                print data['status_msg']
```

Comment:
However I'm attempting to get some human status text (status_msg) from Duo out of the runner that can be shown to the user in the alias without extras, 

From [st2common/log.py](https://github.com/StackStorm/st2/blob/master/st2common/st2common/log.py) all logging methods are decorated and sent to stderror.

When an action succeeds we can `return` a dict and access the data in the result in the alias or publish in a workflow. However if we need an action to fail we don't seem to return data. So this is the first comprise that I've come up with.

May be the action should should just return the data from Duo and leave the processing of the status to the workflow it would be contained in. And it only fail if there was some underling error. However the succeed for being authenticated and failing for being denied would a visual representation in flow.